pkg/shares/share_splitting.go

```diff
@@ -58,6 +58,9 @@ func Split(data coretypes.Data, useShareIndexes bool) ([]Share, error) {
 		return nil, err
 	}
 	currentShareCount += len(blobShares)
+	if currentShareCount > wantShareCount {
+		return nil, fmt.Errorf("too many shares: %d > %d", currentShareCount, wantShareCount)
+	}
```

Comment:
what happens if this is reached during PrepareProposal?

Reply:
> what happens if this is reached during PrepareProposal?

that would indicate an error in https://github.com/celestiaorg/celestia-app/blob/1af66978dac3fa3133dbd6f9be4c60d5642d846e/app/prepare_proposal.go#L25

we could try to reproduce such an error by fuzz testing `estimateSquareSize`

Reply:
yeah, but how do we handle that error?

Reply:
something like https://github.com/rootulp/celestia-app/blob/57126cc55babd643eb73be3dca7a962d8e4e0d25/app/prepare_proposal.go#L59-L68

```golang
	// estimate the square size. This estimation errors on the side of larger
	// squares but can only return values within the min and max square size.
	squareSize, totalSharesUsed, err := estimateSquareSize(parsedTxs)
	if err != nil {
		app.Logger().Error(
			"failure to estimate square size while creating a proposal block",
			"error",
			err.Error(),
		)
		panic(err)
	}
```